app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class ReservationsController < ApplicationController
+  def index
+    @reservations = Reservation.all
+    render json: @reservations
+  end
+
+  def show
+    render json: @reservation
```

Comment:
You need to set the `@reservation` variable before using it. You have the same in `destroy` action. You can use before_action to set the `@reservation` variable for specific actions.

Reply:
would be also a good option to add to my index method line like:
`@reservation  = reservation.find(params[:id])` ?
or maybe create initialize and there:
```
def initialize  
@reservation = reservation.find(params[:id])
end 

```

Reply:
You don't need `@reservation` in the index method(action). You're rendering a collection of `@reservations` there.

Reply:
@baniak66 Ok, I thought also about defining it in show and destroy methods, but I have to DRY. 
I'm not sure how to set up variable to before_action. Should it looks like:
```
before_action :set_reservation 

#some code...

private
def set_reservation
@reservation = reservation.find(params[:id])
end
```

Reply:
That looks ok👍  But please also specify actions that before_action will apply to, like this.
```
before_action :set_reservation, only: [:show, :destroy]
```
Otherwise, before_action will be applied to all controller actions.

Reply:
@baniak66 Thanks !